src/java/org/apache/cassandra/config/DurationSpec.java

```diff
@@ -27,24 +28,25 @@
 import com.google.common.primitives.Ints;
 
 /**
- * Represents a positive time duration.
+ * Represents a positive time duration. Wrapper class for Cassandra duration configuration parameters, providing to the
+ * users the opportunity to be able to provide config with a unit of their choice in cassandra.yaml as per the available
+ * options. (CASSANDRA-15234)
  */
-public final class Duration
+public final class DurationSpec
 {
     /**
      * The Regexp used to parse the duration provided as String.
      */
-    private static final Pattern TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+)([a-zA-Z]{1,2}|µs|µS)$"));
-    private static final Pattern DOUBLE_TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+\\.\\d+)([a-zA-Z]{1,2}|µs|µS)$"));
-    
-    private final long quantity;
+    private static final Pattern TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+)(d|h|s|ms|us|µs|ns|m)"));
+    private static final Pattern DOUBLE_TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+\\.\\d+)(d|h|s|ms|us|µs|ns|m)$"));
```

Comment:
```suggestion
    private static final Pattern DOUBLE_TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+\\.\\d+)(ms|us|µs|ns|m)$"));
```

I'd definitely not allow anything smaller than a ms here. It would make some really strange combinations possible.

Reply:
You mean bigger I think. But then there are parameters like tracetype_query_ttl=1d and trace_type_repair_ttl=7d where it make sense instead of converting to long numbers. You are concerned that operators might mix the units and put something weird? I would expect them to easily mix ms,ns,s but not the minutes, days, maybe I am too positive :D

Reply:
Indeed I meant bigger. Apologies.

Reply:
Essentially, for things on the order of days, I would disallow `double` and just make people specify things in hours, rather than a.) rounding down at large granularities or b.) trying to do more complex conversions of things like `1.5d`.

Reply:
Issue solved, this was a bug in `Duration` committed in a ticket for ttls... those also shouldn't be half day, etc. As I mentioned in another comment, the new spec should allow (for now there is no reason to do it in a different way IMHO) only long and double ms should be used only in the converter for backward compatibility. I added a commit, I will push it with the review comments addressed.